src/Compilers/Core/Portable/MemberDescriptor.cs

```diff
@@ -44,15 +45,35 @@ internal enum MemberFlags : byte
         /// (either for the VB runtime classes, or types like System.Task etc.) will need 
         /// to use IDs that are all mutually disjoint. 
         /// </summary>
-        public readonly short DeclaringTypeId;
+        private readonly short _declaringTypeId;
+
+        public bool IsSpecialTypeMember => _declaringTypeId < (int)InternalSpecialType.NextAvailable;
+
+        public ExtendedSpecialType DeclaringSpecialType
+        {
+            get
+            {
+                Debug.Assert(_declaringTypeId < (int)InternalSpecialType.NextAvailable);
+                return (ExtendedSpecialType)_declaringTypeId;
+            }
+        }
+
+        public WellKnownType DeclaringWellKnownType
+        {
+            get
+            {
+                Debug.Assert(_declaringTypeId >= (int)WellKnownType.First);
+                return (WellKnownType)_declaringTypeId;
+            }
+        }
 
         public string DeclaringTypeMetadataName
         {
             get
             {
-                return DeclaringTypeId <= (int)SpecialType.Count
-                           ? ((SpecialType)DeclaringTypeId).GetMetadataName()!
-                           : ((WellKnownType)DeclaringTypeId).GetMetadataName();
+                return _declaringTypeId < (int)InternalSpecialType.NextAvailable
+                           ? ((ExtendedSpecialType)_declaringTypeId).GetMetadataName()!
+                           : ((WellKnownType)_declaringTypeId).GetMetadataName();
             }
```

Comment:
Since we now have properties for this:
```suggestion
                return IsSpecialTypeMember
                           ? DeclaringSpecialType.GetMetadataName()!
                           : DeclaringWellKnownType.GetMetadataName();
            }
```